### PR TITLE
QMPlay2: new port in multimedia, Qt-based player

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Let’s see if this builds :) 

UPD. _Yay, it does._

I actually got it to work on PowerPC – not just building, but with video playback working.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1
Xcode 15.1

macOS 10A190
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: multimedia/QMPlay2/Portfile [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                    port:jemalloc
+        configure.args-replace \
+                    -DUSE_JEMALLOC=OFF -DUSE_JEMALLOC=ON
+    }
+} else {
+    PortGroup       qt5 1.0
+
+    qt5.depends_component qtdeclarative qttools qttranslations
+
+    github.setup    zaps166 QMPlay2 da6036373a72eab4c346b09e77207c94a35a6cd0
```
Comment (reneeotten) [inline multimedia/QMPlay2/Portfile:64]: I am starting to feel like a broken record.... released versions only please. You know how it works by now so please just follow the guidelines. The Qt5 versions for sure can be the latest release; for the Qt4 compatible version just use the latest release that supports it.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: multimedia/QMPlay2/Portfile (hunk 1 above)
Comment (barracuda156) [reply multimedia/QMPlay2/Portfile:64]: @reneeotten There is no release for that branch. It is a specific branch for Qt4, which upstream explicitly advised to use.
Let me do ppc support in an optimal way, please, and not pick a totally random release and then add 50+ patches on top just to satisfy an arbitrary requirement :)

Comment (barracuda156) [reply multimedia/QMPlay2/Portfile:64]: For Qt5 that is exactly the commit of the latest release. For w/e reason tarball did not want to download for me. I will gladly change it from commit to a tag for Qt5, I just could not make it work.

**Comment by barracuda156** [comment]
@reneeotten I have fixed download from releases for Qt5 version. Perhaps either a network issue yesterday or after spending 20 hrs on this I could not write a version number correctly )
Should work now.

**Comment by barracuda156** [comment]
@reneeotten Could you help merging it now, as from_releases works?

P. S. For Qt4 version, upstream advised to use the branch specifically: https://github.com/zaps166/QMPlay2/issues/674#issuecomment-1868287879
(It already took me literally hours to backport FFMpeg fixes and revert some code which fails, it would be a pain to move patches to another base, and then backport whatever upstream added in the branch.)

**Comment by pmetzger** [comment]
Should this now be merged?

**Comment by barracuda156** [comment]
> Should this now be merged?

@pmetzger I believe so. It works, both on new OS and old. (For the old ones it may be a single player to actually work now.)

**Pull request merged**